modules/apps/commerce/commerce-health-status-web/src/main/java/com/liferay/commerce/health/status/web/internal/util/CommerceServiceAccessPolicyHealthHttpStatus.java

```diff
@@ -0,0 +1,201 @@
+import com.liferay.portal.language.LanguageResources;
+import com.liferay.portal.security.service.access.policy.model.SAPEntry;
+import com.liferay.portal.security.service.access.policy.service.SAPEntryLocalService;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.ResourceBundle;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Luca Pellizzon
+ */
+@Component(
+	enabled = false, immediate = true,
+	property = {
+		"commerce.health.status.display.order:Integer=50",
+		"commerce.health.status.key=" + CommerceHealthStatusConstants.SAP_COMMERCE_HEALTH_STATUS_KEY
+	},
+	service = CommerceHealthHttpStatus.class
+)
+public class CommerceServiceAccessPolicyHealthHttpStatus
+	implements CommerceHealthHttpStatus {
+
+	@Override
+	public void fixIssue(HttpServletRequest httpServletRequest)
+		throws PortalException {
+
+		long companyId = _portal.getCompanyId(httpServletRequest);
+
+		_removeCommerceDefaultSAPEntries(companyId);
+
+		User user = _userLocalService.getDefaultUser(companyId);
+
+		_addCommerceDefaultSAPEntries(companyId, user.getUserId());
+	}
+
+	@Override
+	public String getDescription(Locale locale) {
+		ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
+			"content.Language", locale, getClass());
+
+		return LanguageUtil.get(
+			resourceBundle,
+			CommerceHealthStatusConstants.
+				SAP_COMMERCE_HEALTH_STATUS_DESCRIPTION);
+	}
+
+	@Override
+	public String getKey() {
+		return CommerceHealthStatusConstants.SAP_COMMERCE_HEALTH_STATUS_KEY;
+	}
+
+	@Override
+	public String getName(Locale locale) {
+		ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
+			"content.Language", locale, getClass());
+
+		return LanguageUtil.get(
+			resourceBundle,
+			CommerceHealthStatusConstants.SAP_COMMERCE_HEALTH_STATUS_KEY);
+	}
+
+	@Override
+	public int getType() {
+		return CommerceHealthStatusConstants.
+			COMMERCE_HEALTH_STATUS_TYPE_VIRTUAL_INSTANCE;
+	}
+
+	@Override
+	public boolean isFixed(long companyId, long commerceChannelId)
+		throws PortalException {
+
+		List<SAPEntry> companySAPEntries =
+			_sapEntryLocalService.getCompanySAPEntries(
+				companyId, QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+		for (SAPEntry sapEntry : companySAPEntries) {
+			if (Objects.equals(
+					sapEntry.getName(),
+					CommerceSAPConstants.COMMERCE_SAP_ENTRY_NAME)) {
+
+				String[] sapEntryObjectArray =
+					CommerceSAPConstants.SAP_ENTRY_OBJECT_ARRAYS[0];
+
+				String allowedServiceSignatures = sapEntryObjectArray[1];
+
+				List<String> allowedServiceSignaturesList =
+					sapEntry.getAllowedServiceSignaturesList();
+
+				String[] split = allowedServiceSignatures.split("\n");
+
+				for (String s : split) {
+					if (!allowedServiceSignaturesList.contains(s)) {
+						return false;
+					}
+				}
```

Comment:
We should rename this, maybe like this.
```suggestion
				String[] allowedServiceSignaturesArray = allowedServiceSignatures.split("\n");

				for (String allowedServiceSignature : allowedServiceSignaturesArray ) {
					if (!allowedServiceSignaturesList.contains(allowedServiceSignature)) {
						return false;
					}
				}
```